### PR TITLE
Add tests for operations with tags

### PR DESCRIPTION
**Pull request by otulach**
First test, checking the connection to mocked-gitlab via request POST.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            "name": "Python Debugger: Current File with Arguments",
+            "type": "debugpy",
+            "request": "launch",
+            "program": "${file}",
+            "console": "integratedTerminal",
+            "args": "${command:pickArgs}"
+        }
+    ]
+}
```
Comment (vhotspur) [inline .vscode/launch.json:24]: Would you, please, drop this file from the pull request completely (i.e., please, do an interactive rebase and remove it from the history)? Thanks!

I am not 100% convinced these files should be part of the repository at all (and honestly, M$ again decided to break standards again by calling it JSON and adding a non-standard comments into it).

File: tests/test_createtag.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import logging
+
+import teachers_gitlab.main as tg
+
+def test_create_tag(mock_gitlab):
```
Comment (vhotspur) [inline tests/test_createtag.py:5]: This one looks good :+1:

File: tests/test_createtag.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    
+    tg.action_create_tag(
+        mock_gitlab.get_python_gitlab(),
+        logging.getLogger("createtag"),
+        tg.ActionEntries(entries),
+        'student/{login}',
+        'tag2',
+        'double',
+        ''
+    )
```
Comment (vhotspur) [inline tests/test_createtag.py:81]: We are mocking everything - is it really needed to actually call `tg.action_create_tag` twice to emulate that the tag already exists? But maybe I am missing something very obvious here...

File: tests/test_protecttag.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    mock_gitlab.report_unknown()
+    
+    tg.action_protect_tag(
+        mock_gitlab.get_python_gitlab(),
+        logging.getLogger("protecttag"),
+        tg.ActionEntries(entries),
+        'student/{login}',
+        'tag1',
+        'devel'
+    )
```
Comment (vhotspur) [inline tests/test_protecttag.py:36]: Please, terminate the files with a newline (it is much nicer in the terminal when you `cat` them and the prompt starts on a new line). Thanks!

**Review by vhotspur — CHANGES_REQUESTED** [review]
Only tiny bits, overall this looks very nice! Thanks a lot!

File: tests/test_protecttag.py (hunk 4 above)
Comment (otulach) [reply tests/test_protecttag.py]: No worries, you can check the latest commit now a see if the changes are up to your expectations.

**Comment by vhotspur** [comment]
Looking at the latest commit, it seems to me that multiple tests are named the same? I do not think that could work well and generally it would be confusing.

Feel free to name the tests with longer names (e.g., `test_create_tag_that_does_not_exist` or `test_create_tag_that_already_exists` are not that bad names).

**Review by vhotspur — APPROVED** [review]
LGTM, thanks!

**Pull request merged**